### PR TITLE
Adds the sanity damage reduction component for clothing , adds it to balaclavas

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About The Pull Request
Makes balaclavas provide 60% more protection from environmental sanity damage.
Adds a component for this behaviour that can be added to any item
(Important note , this uses a multiplier , not added on directly , if you have a perk that sets your environment coeff to 0 , it won't help you at all)
Also im a dumbass and i fixed the clothing dropped signals,  because i forgot a case in which the item gets moved to another slot.

## Why It's Good For The Game
Players should be incentivized to wear something other than ciggaretes , vape masks , or drink chemicals every 10 minutes in order to preserve their sanity.
Also its badass to be more mentally tough if you wear a balaclava.
## Changelog
:cl:
add: Added the sanity_damage_reduction component for clothing
add: Made balaclavas and any subtype of gasmasks reduce sanity damage fron environment by 60% 
add:Made surgical masks and a few other masks reduce enviromental damage by 30%
fix: Clothing signals not working as expected.
/:cl:

### CODE REVIEW

File: code/modules/tips_and_tricks/gameplay.dm [hunk 1]
```diff
@@ -8,6 +8,9 @@
 /tipsAndTricks/gameplay/fightBlob
     tipText = "If you're trying to take on a blob, a flamethrower is your best weapon."
 
+/tipsAndTricks/gameplay/maskProtections
+    tipText = "Balaclavas, robber mask and other masks provide a buff to your sanity protection, better wear them than smoke ciggaretes."
```
Comment (Firefox13) [inline code/modules/tips_and_tricks/gameplay.dm:12]:
```suggestion
    tipText = "Balaclavas, robber mask and other masks provide a buff to your sanity protection, better wear them than smoke cigarettes."
```

File: code/datums/components/clothing_sanity_protection.dm [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+// Remove any references to avoid hard dels
+/datum/component/clothing_sanity_protection/Destroy()
+	if(current_user)
+		current_user.sanity.environment_cap_coeff /= environment_cap_buff
+		UnregisterSignal(current_user, COMSIG_CLOTH_DROPPED)
+		current_user = null
+	UnregisterSignal(parent, COMSIG_CLOTH_EQUIPPED)
+	var/atom/current_parent = parent
+	current_parent.description_info -=  "This item reduces sanity damage taken from enviromental factors \n"
```
Comment (Firefox13) [inline code/datums/components/clothing_sanity_protection.dm:33]:
```suggestion
	current_parent.description_info -=  "This item reduces sanity damage taken from enviromental factors. \n"
```

File: code/datums/components/clothing_sanity_protection.dm [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/datum/component/clothing_sanity_protection
+	var/environment_cap_buff
+	var/mob/living/carbon/human/current_user
+
+/datum/component/clothing_sanity_protection/Initialize(value)
+	if(!(istype(parent, /obj/item/clothing)))
+		return COMPONENT_INCOMPATIBLE
+	environment_cap_buff = value
+	var/atom/current_parent = parent
+	current_parent.description_info += "This item reduces sanity damage taken from enviromental factors \n"
```
Comment (Firefox13) [inline code/datums/components/clothing_sanity_protection.dm:10]:
```suggestion
	current_parent.description_info += "This item reduces sanity damage taken from environmental factors. \n"
```

**Review by TheShown911 — APPROVED** [review]
This is such a stupid video game logic moment it goes back to being smart, I approve.

**Pull request merged**